fvtest/porttest/omrsockTest.cpp

```diff
@@ -757,3 +757,253 @@ TEST(PortSockTest, socket_flags)
+	EXPECT_EQ(OMRPORTLIB->sock_addrinfo_length(OMRPORTLIB, &result, &length), 0);
+	ASSERT_NE(length, uint32_t(0));
+
+	/* Create a nonblocking socket with results */
+	for (uint32_t i = 0; i < length; i++) {
+		if (0 == OMRPORTLIB->sock_socket(OMRPORTLIB, &clientSocket, OMRSOCK_AF_INET, OMRSOCK_STREAM | OMRSOCK_O_NONBLOCK, OMRSOCK_IPPROTO_DEFAULT)) {
+			ASSERT_NE(clientSocket, (void *)NULL);
+			EXPECT_EQ(OMRPORTLIB->sock_addrinfo_address(OMRPORTLIB, &result, i, &clientSockAddr), 0);
+			ASSERT_EQ(OMRPORTLIB->sock_bind(OMRPORTLIB, clientSocket, &clientSockAddr), 0);
+			break;
+		}
+	}
+	EXPECT_EQ(OMRPORTLIB->sock_freeaddrinfo(OMRPORTLIB, &result), 0);
+
+	/* Client Attempts Connection */
+	while (0 != (rc = OMRPORTLIB->sock_connect(OMRPORTLIB, clientSocket, &serverSockAddr))) {
+		if (OMRPORT_ERROR_SOCKET_INPROGRESS == rc) {
+			// Still in the middle of attempting connection.
+			continue;
+		}
+		else if (OMRPORT_ERROR_SOCKET_ALREADY_CONNECTED == rc) {
+			// Connected
+			break;
+		}
+		else {
+			FAIL() << "Error other than INPROGRESS and ALREADY_CONNECTED were returned for connect: " << ::testing::PrintToString(rc);
+		}
+	}
+
+	/* Server Accepts Connection */
+	OMRSockAddrStorage connectedServerSockAddr;
+	omrsock_socket_t connectedServerSocket = NULL;
+	ASSERT_EQ(OMRPORTLIB->sock_accept(OMRPORTLIB, serverSocket, &connectedServerSockAddr, &connectedServerSocket), 0);
+
+	ASSERT_EQ(OMRPORTLIB->sock_close(OMRPORTLIB, &connectedServerSocket), 0);
+	ASSERT_EQ(OMRPORTLIB->sock_close(OMRPORTLIB, &clientSocket), 0);
+	ASSERT_EQ(OMRPORTLIB->sock_close(OMRPORTLIB, &serverSocket), 0);
+}
+
+/**
+ * Test nonblocking functionality by two nonblocking socket communication.
+ *
+ * First, a connection is established between server and client. The server
+ * and client is then set to be nonblocking. Communication between server and
+ * client is then tested using OMRPORT_ERROR_SOCKET_WOULDBLOCK.
+ */
+TEST(PortSockTest, two_socket_nonblock_communication_basic)
+{
+	OMRPORT_ACCESS_FROM_OMRPORT(portTestEnv->getPortLibrary());
+
+	OMRSockAddrStorage serverSockAddr;
+	omrsock_socket_t serverSocket = NULL;
+	OMRSockAddrStorage clientSockAddr;
+	omrsock_socket_t clientSocket = NULL;
+	OMRSockAddrStorage connectedServerSockAddr;
+	omrsock_socket_t connectedServerSocket = NULL;
+	uint16_t port = 4930;
+	uint32_t inaddrAny;
+	uint8_t serverAddr[4];
+
+	/* To Create a Server Socket and Address */
+	inaddrAny = OMRPORTLIB->sock_htonl(OMRPORTLIB, OMRSOCK_INADDR_ANY);
+	memcpy(serverAddr, &inaddrAny, 4);
+	EXPECT_EQ(OMRPORTLIB->sock_sockaddr_init(OMRPORTLIB, &serverSockAddr, OMRSOCK_AF_INET, serverAddr, OMRPORTLIB->sock_htons(OMRPORTLIB, port)), 0);
+	start_server(OMRPORTLIB, OMRSOCK_AF_INET, OMRSOCK_STREAM, &serverSocket, &serverSockAddr);
+
+	/* To Create a Client Socket and Address */
+	connect_client_to_server(OMRPORTLIB, (char *)"localhost", NULL, OMRSOCK_AF_INET, OMRSOCK_STREAM, &clientSocket, &clientSockAddr, &serverSockAddr);
```

Comment:
why do you need the cast to `char*` here?

Reply:
I will remove them.